keras/backend/tensorflow_backend.py

```diff
@@ -3942,8 +3942,8 @@ def ctc_batch_cost(y_true, y_pred, input_length, label_length):
         Tensor with shape (samples,1) containing the
             CTC loss of each element.
     """
-    label_length = tf.to_int32(tf.squeeze(label_length))
-    input_length = tf.to_int32(tf.squeeze(input_length))
+    label_length = tf.to_int32(tf.squeeze(label_length, axis=1))
+    input_length = tf.to_int32(tf.squeeze(input_length, axis=1))
```

Comment:
`axis=1` is the first axis, not the last axis (-1).

Please add a unit test.

Reply:
@fchollet 
Both original inputs `label_length` and `input_length` are **2D** tensors with shape (samples, 1) and the index of `axis` starts at 0, so I think either `axis=1` or `axis=-1` refers to the same axis.

Reply:
Sure, but for the sake of semantics and possible future extensibility, please use `-1` to refer to the last axis.

Reply:
Okay.
By the way, should I add a unit test at `test_ctc()` in `backend_test.py`, right?

Reply:
You can add a test case in https://github.com/keras-team/keras/blob/master/tests/keras/backend/backend_test.py#L1446

Or you can just add a new function.